stack_image.sh

```diff
@@ -0,0 +1,305 @@
+            if test $# -gt 0; then
+                LOGENTRIES_TOKEN=$1
+            else
+            	usage_exit "No token given."
+            fi
+			shift
+		    ;;
+		-lu|--logentries-url)
+		    shift
+            if test $# -gt 0; then
+                LOGENTRIES_URL=$1
+            else
+            	usage_exit "No url given."
+            fi
+			shift
+		    ;;
+		-lp|--logentries-port)
+		    shift
+            if test $# -gt 0; then
+                LOGENTRIES_PORT=$1
+            else
+            	usage_exit "No port given."
+            fi
+			shift
+		    ;;
+		*)
+			break
+			;;
+	esac
+done
+
+REMAINING=$@
+
+# Check that input file is set
+if [[ "AAA${INPUT_FILE}AAA" == "AAAAAA" ]]; then
+	usage_exit "No input file defined. (mandatory)"
+fi
+
+# Split input files
+IFS=',' read -ra INPUT_FILES <<< "$INPUT_FILE"
+
+echo $INPUT_FILES
+
+# Check if URL input, convert supports it natively but would download every time is called. Download here.
+URL_REGEX='(https?|ftp|file)://[-A-Za-z0-9\+&@#/%?=~_|!:,.;]*[-A-Za-z0-9\+&@#/%=~_|]'
+
+for ((i=0; i<${#INPUT_FILES[@]}; i++)); do
+    INPUT_FILE="${INPUT_FILES[$i]}"
+
+    # @todo should check that the provided filename are unique
+
+    if [[ ${INPUT_FILE} =~ ${URL_REGEX} ]]; then
+        URL=${INPUT_FILE}
+        INPUT_FILE=$(basename "$(echo ${URL} | cut -d@ -f2 | cut -d/ -f2- | cut -d? -f1)")
+        ${WGET} -q -O "${INPUT_FILE}" "${URL}"
+    fi
+
+    # Sanitize input file (remove special chars, convert spaces to underscore")
+    INVALID_REGEX='[ :?"\\]'
+    if [[ ${INPUT_FILE} =~ ${INVALID_REGEX} ]]; then
+        SANITIZED_INPUT=${INPUT_FILE//[:?\']}
+        SANITIZED_INPUT=${SANITIZED_INPUT// /_}
+
+        # @todo: Should check that does not already exists
+
+        # Create symbolic link so that there is no need to copy or move data
+        ln -f -s "${INPUT_FILE}" $SANITIZED_INPUT
+
+        INPUT_FILES[$i]=$SANITIZED_INPUT
+        CLEANUP_FILES+=" $SANITIZED_INPUT"
+    fi
+done
+
+# Check that output file is set
+if [[ "AAA${OUTPUT_FILE}AAA" == "AAAAAA" ]]; then
+	usage_exit "No output file defined. (mandatory)"
+fi
+
+# Check column modes
+case "$COLS_MODE" in
+	rl|rtl|right-to-left)
+	    COL_MERGE_CMD="+append"
+		;;
+	tb|ttb|top-to-bottom)
+		COL_MERGE_CMD="-append"
+        ;;
+    *)
+        usage_exit "Invalid append mode, must be \"right-to-left\" or \"top-to-bottom\"."
+    	;;
+esac
+
+
+function log() {
+    if [[ ! "AAA${LOGENTRIES_TOKEN}AAA" == "AAAAAA" ]]; then
+        echo "${LOGENTRIES_TOKEN} $@" | telnet $LOGENTRIES_URL $LOGENTRIES_PORT >/dev/null 2>&1
+    fi
+}
+
+### END CONFIGURATION, BEGIN WORK ###
+
+# Override columns if more than one input file is given
+if [[ ${#INPUT_FILES[@]} -gt 1 ]]; then
+	COLUMNS=${#INPUT_FILES[@]}
+fi
+
+
+## Check that input file is present
+#if [[ ! -r ${INPUT_FILE} ]]; then
+#    echo "Input file \"${INPUT_FILE}\" not readable."
+#    log "ERROR: Input file \"${INPUT_FILE}\" not readable."
+#    exit 1;
+#fi
```

Comment:
Test has to be done, do as a function and use on column creation, code just above